src/gpodder/model.py

```diff
@@ -895,12 +938,24 @@ def get_play_info_string(self, duration_only=False):
 
     def update_from(self, episode):
         for k in ('title', 'url', 'episode_art_url', 'description', 'description_html', 'chapters', 'link',
-                  'published', 'guid', 'payment_url'):
+                  'published', 'guid', 'payment_url', 'episode_num', 'season', 'author'):
             setattr(self, k, getattr(episode, k))
         # Don't overwrite file size on downloaded episodes
         # See #648 refreshing a youtube podcast clears downloaded file size
         if self.state != gpodder.STATE_DOWNLOADED:
             setattr(self, 'file_size', getattr(episode, 'file_size'))
+    
+    @property
+    def cover_file(self):
+        #check if there is an episode cover in the first place
+        if self.episode_art_url is None:
+            return None
+        #create a cover file name from the episode_art_url
```

Comment:
Three issues there:
1. covername should be sanitized + truncated before using as a filename
2. maybe it could be prefixed with `"cover_"` or something so we can distinguish it from downloaded images from image feeds (not that they are much used (never used them personally) but we try)? See `check_download_folder-)` where any .jpg or .png is prevented to be recognised as external download now. Or could it be named '"cover_<episode.guid>.art" and then be really easy to find? The extension is blindly taken from the URL. Is it always .png or .jpg? No .gif, .webp? Maybe studying the channel artwork would give ideas to implement this...
3. Is episode art deleted when we delete an episode?

Reply:
I honestly would actually like some help in this, since this is kind of a messy solution and im 99% sure it could be done better. right now if the podcast owner is messy with their filenames you end up with "duplicated" pictures that look the same and are just different names. not usre if there is an easy way around this other than downloading and then doing a comparison check ***every time*** which would be kind of slow. also I dont believe it is deleted when an episode is deleted as I didnt add any check in for this. someone with a better idea of a implementation here might want to take a stab
![same pictures](https://user-images.githubusercontent.com/32679887/209964798-80009374-2b33-43d3-abb4-4897406585e6.PNG)

Reply:
I already have a working episode thumbnail download queueing system that I've been using for 1.5 years now. I will try to get it polished up and a PR opened in the next month.

It creates a `thumbnails` folder inside each channel downloads folder to separate them. Each thumbnail is `{episode ID}.{ext}`, and yes, the youtube-dl extension produces webp files for youtube channels. And the default thumbnail for built-in youtube support has black bars, so my system uses a different URL.

I do have a feed which uses the same URL for every episode, and need to detect when an URL is queued multiple times so the download only happens once. Also not implemented yet, but the ETag could be stored for last downloaded thumbnail. This would copy the local file, if one still exists, when downloading the same URL the following week. The Etag should be the same for your feed which has different URLs for the same thumbnail, and this would only download them once.

Reply:
not sure I understand the last part, but the issue is here that each season has a different thumbnail, so its doing that correctly, just that their is multiple files with the same seasons thumbnail just a different name. Could possibly setup the name to include the season then if the seasons thumbnail is already downloaded it ignores it? that only falls apart when podcasts dont use seasons but still have changing thumbs....

Reply:
Yes, good idea to have a separate folder for thumbnails.

Reply:
so what should I do about this? Should I leave it in to be fixed in a later thumbnails implementation or should I remove it to be re added later?

Reply:
The patch doesn't work without it, so leave it in for now.